src/intrinsics.cpp

```diff
@@ -512,7 +512,7 @@ static jl_cgval_t generic_bitcast(jl_codectx_t &ctx, const jl_cgval_t *argv)
     if (!jl_is_primitivetype(v.typ) || jl_datatype_size(v.typ) != nb) {
         Value *typ = emit_typeof_boxed(ctx, v);
         if (!jl_is_primitivetype(v.typ)) {
-            if (jl_is_datatype(v.typ) && !jl_is_abstracttype(v.typ)) {
+            if (!isboxed) {
```

Comment:
I think this was missing a negation instead? In general, depending on the result of julia_type_to_llvm is unreliable

Reply:
It is trying to check if there may exist a subtype of `v.typ` that is a primitivetype, since otherwise we can emit an unconditional error. A lot of our code assumes that `bottom` is already handled though, and will probably fail if that gets through. Usually we do something equivalent to this after any `emit_expr` call, so that the rest of the code does not need to handle this case:
```
         if (argv[i].typ == jl_bottom_type)
             return jl_cgval_t();
```